### PR TITLE
Fix inconsistent behavior of getCardSize for weather forecast

### PR DESCRIPTION
**Pull request by mback2k**
## Proposed change

Make getCardSize() and render() behave identical by using the
same logic to retrieve the weather and/or forecast condition.

Follow up to #10267

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

See #10267

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by zsarnett — COMMENTED** [review]
For the getCardSize Logic. We only really need to see if the config is there and the stateObj has a forecast on it. 

Now we are also slicing the array when we do not need to

**Comment by mback2k** [comment]
Good point, will try to optimize it further.

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts [hunk 1]
```diff
@@ -87,11 +87,13 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
   }
 
   public getCardSize(): number {
+    const [weather, forecast] = this._prepareRenderData();
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-weather-forecast-card.ts:90]: `getCardSize` should only use config, and not state info

Comment (mback2k) [reply src/panels/lovelace/cards/hui-weather-forecast-card.ts:90]: Then this PR has no purpose as an alignment would require state and the existing behavior is the only solution.

**Comment by mback2k** [comment]
The existing behavior is not a big problem as the config editor already checks that a forecast is actually available on the selected entity. I guess this can be closed then?

**Pull request closed without merging**